include/boost/redis/connection.hpp

```diff
@@ -1281,15 +1274,19 @@ class connection {
    }
 
    /// Calls `boost::redis::basic_connection::async_exec`.
-   template <class Response, class CompletionToken>
-   auto async_exec(request const& req, Response& resp, CompletionToken&& token)
+   template <class Response, class CompletionToken = asio::deferred_t>
```

Comment:
You can also consider using `BOOST_ASIO_COMPLETION_TOKEN_FOR` to implement concept checks.